File: C++/Hangman/Hangman.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <sstream>
#include <algorithm>
#include <vector>
#include <ctime>

using namespace std;

char asciitolower(char c);

int main() {
    string line;
    vector<string> strings;
    ifstream myfile;
    string path = "C:/Users/nikla/Desktop/JavaExample/depend/list.txt", input;
    cout << "Write the path to the list and after that next. Example: C:/Users/nikla/Desktop/JavaExample/depend/list.txt" << endl;
    while (getline(cin, input)) {
        if (input == "next")
            break;
        path = input;
        cout << "You set the path to: " << path << endl;
    }

    //getting the file
    myfile.open(path);
    if (myfile.is_open()) {
        int i = 0;
        //scanning the file
        while (getline(myfile, line)) {

            transform(line.begin(), line.end(), line.begin(), asciitolower);
            strings.push_back(line);
            i++;
        }
    } else {
        cout << "Unable to find the file!" << endl;
        system("pause");
    }

    srand(static_cast<unsigned int>(time(nullptr)));

    int wrong = 0, max = 10;
    int r = rand() % (strings.size() - 0 + 1) + 0;
    string guess, c = strings[r];
    cout << "Start guessing!" << endl;
    cout << "The word is " << c.length() << " characters long." << endl;
    string pattern;
    int i = 0;
    vector<char> guessed = {};
    vector<string> allGuessed = {};

    for (int k = 0; k < c.length(); k++) {
        pattern.append("_");
    }

    cout << pattern << endl;

    //game loop
    while (true) {
            cin >> guess;
        if (find(allGuessed.begin(), allGuessed.end(), guess) != allGuessed.end()) {
            cout << "You already guessed that character/string!" << endl;
            continue;
        }
        if (c == guess && guess.length() > 1) {
            cout << "You guessed to word! The word was " << c << "." << endl;
            break;
        } else {
            bool b = false;
            for (char i : c) {
                if (i == guess[0]) {
                    b = true;
                }
            }
            if (b) {
                cout << "The character " << guess[0] << " is part of the word." << endl;
                guessed.push_back(guess[0]);
                pattern = "";
                stringstream ss;
                for (char car : c) {
                    if (car == guess[0] || find(guessed.begin(), guessed.end(), car) != guessed.end()) {
                        ss << car;
                    } else ss << "_";
                }
                pattern = ss.str();
                if (pattern == c) {
                    cout << "You won! You had " << wrong << " wrong guesses out of " << max << " max possible wrong guesses! The word was " << c << "." << endl;
                    break;
                }
                cout << pattern << endl;
            } else {
                wrong++;
                cout << "The character " << guess[0] << " isn't part of the word. You have " << wrong
                     << " wrong guess of " << max << " possible guesses." << endl;
            }
        }
        allGuessed.push_back(guess);
        if (wrong > max - 1) {
            cout << "You are out of guesses." << " The word was " << c << "." << endl;
            break;
        }
    }
    system("pause");
    return 0;
}

char asciitolower(char in) {
    if (in <= 'Z' && in >= 'A')
        return in - ('Z' - 'z');
    return in;
}

File: src/main/java/com/nov0cx/Hangman.java
package com.nov0cx;

import com.nov0cx.utils.ConsoleColor;
import lombok.Getter;
import lombok.Setter;
import lombok.SneakyThrows;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Random;

@Getter
@Setter
public class Hangman {

    @SneakyThrows
    public Hangman() {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));

        //looking for the path of the text-file with the list
        String path = "C:/Users/nikla/Desktop/JavaExample/depend/list.txt";
        System.out.println("Write the path to the list and after that 'next'");
        String line;
        while (true) {
            line = reader.readLine();
            if (line.equals("next"))
                break;
            path = (line);
            System.out.println(ConsoleColor.GREEN + "Set path to the list as " + ConsoleColor.AQUA + line + ConsoleColor.GREEN + "." + ConsoleColor.RESET);
        }

        //adding the list of the file in the list
        File file = new File(path);
        ArrayList<String> strings = new ArrayList<>();
        FileReader reader1 = new FileReader(file);
        BufferedReader reader2 = new BufferedReader(reader1);
        String l;
        while ((l = reader2.readLine()) != null) {
            strings.add(l);
        }

        //start of the game
        String c = strings.get(new Random().nextInt(strings.size() + 1)).toLowerCase();
        System.out.println(ConsoleColor.GREEN + "The word is " + c.toCharArray().length + " characters long." + ConsoleColor.RESET);

        //the number of wrong guesses
        int wrong = 0;
        //the maximum of wrong guesses
        int max = 10;

        //guess chars
        ArrayList<Character> guessed = new ArrayList<>();
        ArrayList<Character> allGuessed = new ArrayList<>();

        int cLength = 0;
        ArrayList<Character> got = new ArrayList<>();
        for (Character car : c.toCharArray()) {
            if (!got.contains(car)) {
                cLength++;
                got.add(car);
            }

        }

        StringBuilder b = new StringBuilder();
        String realPattern = "";
        for(Character car : c.toCharArray()) {
            b.append("_");
        }
        realPattern = b.toString();
        System.out.println(ConsoleColor.GREEN + realPattern + ConsoleColor.RESET);

        //game loop
        while (guessed.size() != cLength) {
            String read = reader.readLine().toLowerCase();
            if (read.length() != 1) {
                System.out.println(ConsoleColor.RED + "You can only guess one character at once!" + ConsoleColor.RESET);
                continue;
            }
            if(allGuessed.contains(read.toCharArray()[0])) {
                System.out.println(ConsoleColor.GREEN + "You already guessed that character." + ConsoleColor.RESET);
                continue;
            }

            if (read.equals(c))
                break;
            if (c.contains(read)) {
                System.out.println(ConsoleColor.GREEN + read + " is part of the word." + ConsoleColor.RESET);
                if (!guessed.contains(read.toCharArray()[0]))
                    guessed.add(read.toCharArray()[0]);
                StringBuilder b2 = new StringBuilder();
                for(Character car : c.toCharArray()) {
                    if(car == read.toCharArray()[0] || guessed.contains(car))
                        b2.append(car);
                    else b2.append("_");
                }
                realPattern = b2.toString();
            } else {
                wrong++;
                System.out.println(ConsoleColor.RED + read
                        + " isn't part of the word. You have " + wrong
                        + " wrong guesses out of max " + max + " wrong guesses." + ConsoleColor.RESET);
            }
            System.out.println(ConsoleColor.GREEN + "You already guessed " + Arrays.toString(guessed.toArray()) + "." + ConsoleColor.RESET);
            System.out.println(ConsoleColor.GREEN + realPattern + ConsoleColor.RESET);
            allGuessed.add(read.toCharArray()[0]);
            if (wrong > max - 1)
                break;
        }
        if (wrong > max - 1)
            System.out.println(ConsoleColor.RED + "You are out of guesses. The word was " + c + "." + ConsoleColor.RESET);
        else
            System.out.println(ConsoleColor.GREEN + "You won! The word was " + c + ". You had " + wrong + " wrong guesses." + ConsoleColor.RESET);
    }

    public static void main(String[] args) {
        new Hangman();
    }


}
